var/spack/repos/builtin/packages/py-future/package.py

```diff
@@ -0,0 +1,37 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class PyFuture(PythonPackage):
+    """Clean single-source support for Python 3 and 2"""
+
+    homepage = "https://python-future.org/"
+    url = "https://pypi.io/packages/source/f/future/future-0.16.0.tar.gz"
+
+    version('0.16.0', '3e8e88a2bda48d54b1da7634d04760d7')
+    version('0.15.2', 'a68eb3c90b3b76714c5ceb8c09ea3a06')
+
+    depends_on('py-setuptools', type='build')
```

Comment:
Hi @adamjstewart ,

Sadly, with `2.6.0:2.6.999` doesn't work either. Just before to leave it commented, is there any way to access to the `spec` at this level, something similar to this:

```
depends_on('py-setuptools', type='build')
if "python@2.6" in self.spec:
       depends_on('py-importlib', type=('build', 'run'))
       depends_on('py-argparse', type=('build', 'run'))
```

It is not the best solution, but something similar can temporarily work till the concretization is fixed. Otherwise I can just send it commented.